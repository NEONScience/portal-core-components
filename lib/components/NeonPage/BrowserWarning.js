"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _universalCookie = _interopRequireDefault(require("universal-cookie"));
var _styles = require("@material-ui/core/styles");
var _Snackbar = _interopRequireDefault(require("@material-ui/core/Snackbar"));
var _SnackbarContent = _interopRequireDefault(require("@material-ui/core/SnackbarContent"));
var _Button = _interopRequireDefault(require("@material-ui/core/Button"));
var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));
var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));
var _Theme = _interopRequireDefault(require("../Theme/Theme"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
/*
  Browser duck typing and warning
  We know detecting browser on user agent strings for feature gating is bad practice.
  Here, though, we just want a generic warning if we know the browser is IE.
  Nothing else is or should be predicated on the value detected here.
  Don't make this any more complicated!
*/
const isBrowserIE = () => navigator.userAgent.indexOf('MSIE') !== -1 || navigator.appVersion.indexOf('Trident/') > -1;
const useStyles = (0, _styles.makeStyles)(theme => ({
  browserWarning: {
    backgroundColor: theme.palette.error.main
  },
  browserWarningMessage: {
    display: 'flex',
    alignItems: 'center'
  },
  buttonRow: {
    marginTop: theme.spacing(1),
    display: 'flex',
    justifyContent: 'space-around'
  }
}));
const cookies = new _universalCookie.default();
const BrowserWarning = () => {
  const classes = useStyles(_Theme.default);
  const browserIsIE = isBrowserIE();
  const [browserWarningOpen, setBrowserWarningOpen] = (0, _react.useState)(browserIsIE);
  if (!browserIsIE) {
    return null;
  }
  if (cookies.get('ignoreIE11Warning')) {
    return null;
  }
  const handleBrowserWarningClose = () => {
    cookies.set('ignoreIE11Warning', true, {
      path: '/',
      maxAge: 86400
    });
    setBrowserWarningOpen(false);
  };
  const buttonProps = {
    color: 'inherit',
    target: '_blank'
  };
  const message = /*#__PURE__*/_react.default.createElement("span", {
    id: "browser-warning",
    className: classes.browserWarningMessage,
    "data-selenium": "neon-page.browser-warning"
  }, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("b", null, "Your browser has limited support on the NEON Data Portal."), /*#__PURE__*/_react.default.createElement("br", null), "Some parts of the portal will not work for you.", /*#__PURE__*/_react.default.createElement("br", null), "Please use a modern browser for full support.", /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.buttonRow
  }, /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, buttonProps, {
    href: "https://www.mozilla.org/en-US/firefox/new/"
  }), "Firefox"), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, buttonProps, {
    href: "https://www.google.com/chrome/"
  }), "Chrome"), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, buttonProps, {
    href: "https://www.apple.com/safari/"
  }), "Safari"), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, buttonProps, {
    href: "https://www.microsoft.com/en-us/windows/microsoft-edge"
  }), "Edge"))));
  return /*#__PURE__*/_react.default.createElement(_Snackbar.default, {
    anchorOrigin: {
      vertical: 'bottom',
      horizontal: 'right'
    },
    ContentProps: {
      'aria-describedby': 'browser-warning'
    },
    open: browserWarningOpen
  }, /*#__PURE__*/_react.default.createElement(_SnackbarContent.default, {
    className: classes.browserWarning,
    message: message,
    action: /*#__PURE__*/_react.default.createElement(_IconButton.default, {
      key: "close",
      "aria-label": "close",
      color: "inherit",
      onClick: handleBrowserWarningClose
    }, /*#__PURE__*/_react.default.createElement(_Close.default, null))
  }));
};
var _default = exports.default = BrowserWarning;