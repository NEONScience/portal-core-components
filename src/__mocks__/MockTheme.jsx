import React from 'react';

import CssBaseline from '@mui/material/CssBaseline';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';

import Theme from '../lib_components/components/Theme/Theme';

const MockTheme = ({ children }) => {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={Theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

export default MockTheme;
