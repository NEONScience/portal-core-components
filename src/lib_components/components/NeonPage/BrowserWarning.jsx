import React, { useState } from 'react';

import Cookies from 'universal-cookie';

import { makeStyles } from '@mui/styles';
import Snackbar from '@mui/material/Snackbar';
import SnackbarContent from '@mui/material/SnackbarContent';
import Button from '@mui/material/Button';
import IconButton from '@mui/material/IconButton';
import CloseIcon from '@mui/icons-material/Close';

import Theme from '../Theme/Theme';

/*
  Browser duck typing and warning
  We know detecting browser on user agent strings for feature gating is bad practice.
  Here, though, we just want a generic warning if we know the browser is IE.
  Nothing else is or should be predicated on the value detected here.
  Don't make this any more complicated!
*/
const isBrowserIE = () => (
  navigator.userAgent.indexOf('MSIE') !== -1
    || navigator.appVersion.indexOf('Trident/') > -1
);

const useStyles = makeStyles((theme) => ({
  browserWarning: {
    backgroundColor: theme.palette.error.main,
  },
  browserWarningMessage: {
    display: 'flex',
    alignItems: 'center',
  },
  buttonRow: {
    marginTop: theme.spacing(1),
    display: 'flex',
    justifyContent: 'space-around',
  },
}));

const cookies = new Cookies();

const BrowserWarning = () => {
  const classes = useStyles(Theme);
  const browserIsIE = isBrowserIE();
  const [browserWarningOpen, setBrowserWarningOpen] = useState(browserIsIE);

  if (!browserIsIE) {
    return null;
  }

  if (cookies.get('ignoreIE11Warning')) {
    return null;
  }

  const handleBrowserWarningClose = () => {
    cookies.set('ignoreIE11Warning', true, { path: '/', maxAge: 86400 });
    setBrowserWarningOpen(false);
  };

  const buttonProps = {
    color: 'inherit',
    target: '_blank',
  };
  const message = (
    <span
      id="browser-warning"
      className={classes.browserWarningMessage}
      data-selenium="neon-page.browser-warning"
    >
      <div>
        <b>Your browser has limited support on the NEON Data Portal.</b>
        <br />
        Some parts of the portal will not work for you.
        <br />
        Please use a modern browser for full support.
        <br />
        <div className={classes.buttonRow}>
          <Button {...buttonProps} href="https://www.mozilla.org/en-US/firefox/new/">
            Firefox
          </Button>
          <Button {...buttonProps} href="https://www.google.com/chrome/">
            Chrome
          </Button>
          <Button {...buttonProps} href="https://www.apple.com/safari/">
            Safari
          </Button>
          <Button {...buttonProps} href="https://www.microsoft.com/en-us/windows/microsoft-edge">
            Edge
          </Button>
        </div>
      </div>
    </span>
  );

  return (
    <Snackbar
      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
      ContentProps={{ 'aria-describedby': 'browser-warning' }}
      open={browserWarningOpen}
    >
      <SnackbarContent
        className={classes.browserWarning}
        message={message}
        action={(
          <IconButton
            key="close"
            aria-label="close"
            color="inherit"
            onClick={handleBrowserWarningClose}
            size="large"
          >
            <CloseIcon />
          </IconButton>
        )}
      />
    </Snackbar>
  );
};

export default BrowserWarning;
