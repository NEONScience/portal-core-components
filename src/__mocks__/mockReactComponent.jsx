/* eslint react/prop-types: 0 */
/**
   Mock React Component

   Because jest.mock calls can't be dynamic / must hard-code an import path we still need individual
   mocks for every React component that requires mocking. This format defines a generic mock
   implementation for any react component that will appear in snapshots as a div with props mapped
   safely to attributes and any children intact.

   Usage:

   --

   import 'path/to/__mocks__/mockReactComponent';

   jest.mock('path/to/Component', () => mockReactComponent('path/to/Component'));
*/

import React from 'react';

export default function mockReactComponent(path) {
  return (props) => {
    const dataProps = {};
    Object.keys(props).forEach((prop) => {
      if (prop === 'children') { return; }
      const value = props[prop];
      let renderableValue = value;
      try {
        if (typeof value === 'object') { renderableValue = JSON.stringify(value); }
        if (typeof value === 'function') { renderableValue = value.toString(); }
      } catch (error) {
        renderableValue = `Non-renderable ${typeof value} (${error})`;
      }
      dataProps[`prop-${prop}`] = renderableValue;
    });
    const { children } = props;
    return children ? (
      <div ComponentPath={path} {...dataProps}>
        {children}
      </div>
    ) : (
      <div ComponentPath={path} {...dataProps} />
    );
  };
}
