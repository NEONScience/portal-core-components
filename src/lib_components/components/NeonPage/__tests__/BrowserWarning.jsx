import React from 'react';
import renderer from 'react-test-renderer';

import Cookies from 'universal-cookie'; // eslint-disable-line no-unused-vars

import MockTheme from '../../../../__mocks__/MockTheme';
import mockReactComponent from '../../../../__mocks__/mockReactComponent';

jest.mock('@mui/material/Snackbar', () => mockReactComponent('@mui/material/Snackbar'));

// eslint-disable-next-line import/first
import BrowserWarning from '../BrowserWarning';

const mockCookieGet = jest.fn();
jest.mock('universal-cookie', () => {
  return jest.fn().mockImplementation(() => ({
    get: () => mockCookieGet(),
    set: jest.fn(),
  }));
});

// Make key attributes of the Navigator object mockable to simulate different IE iterations
['userAgent', 'appVersion'].forEach((attrib) => {
  Object.defineProperty(window.navigator, attrib, ((value) => ({
    get() { return value; },
    set(v) { value = v; }, // eslint-disable-line no-param-reassign
  }))(window.navigator[attrib]));
});

describe('NeonPage - BrowserWarning', () => {
  beforeEach(() => {
    mockCookieGet.mockReset();
    global.navigator.userAgent = 'foo';
    global.navigator.appVersion = 'bar';
  });
  test('renders nothing if browser is not IE', () => {
    const tree = renderer.create(
      <MockTheme>
        <BrowserWarning />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders nothing if browser is found to be IE but ignore cookie is present', () => {
    global.navigator.userAgent = 'foo MSIE qux';
    mockCookieGet.mockReturnValue(true);
    const tree = renderer.create(
      <MockTheme>
        <BrowserWarning />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders warning if browser is found to be IE and ignore cookie is not set', () => {
    global.navigator.appVersion = 'bar Trident/qux';
    mockCookieGet.mockReturnValue(false);
    const tree = renderer.create(
      <MockTheme>
        <BrowserWarning />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
