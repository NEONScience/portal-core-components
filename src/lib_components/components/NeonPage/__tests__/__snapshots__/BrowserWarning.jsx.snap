// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonPage - BrowserWarning renders nothing if browser is found to be IE but ignore cookie is present 1`] = `null`;

exports[`NeonPage - BrowserWarning renders nothing if browser is not IE 1`] = `null`;

exports[`NeonPage - BrowserWarning renders warning if browser is found to be IE and ignore cookie is not set 1`] = `
<div
  ComponentPath="@mui/material/Snackbar"
  prop-ContentProps="{"aria-describedby":"browser-warning"}"
  prop-anchorOrigin="{"vertical":"bottom","horizontal":"right"}"
  prop-open={true}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root makeStyles-browserWarning-7 css-3em1l5-MuiPaper-root-MuiSnackbarContent-root"
    role="alert"
  >
    <div
      className="MuiSnackbarContent-message css-1exqwzz-MuiSnackbarContent-message"
    >
      <span
        className="makeStyles-browserWarningMessage-8"
        data-selenium="neon-page.browser-warning"
        id="browser-warning"
      >
        <div>
          <b>
            Your browser has limited support on the NEON Data Portal.
          </b>
          <br />
          Some parts of the portal will not work for you.
          <br />
          Please use a modern browser for full support.
          <br />
          <div
            className="makeStyles-buttonRow-9"
          >
            <a
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1tvghku-MuiButtonBase-root-MuiButton-root"
              href="https://www.mozilla.org/en-US/firefox/new/"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              target="_blank"
            >
              Firefox
            </a>
            <a
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1tvghku-MuiButtonBase-root-MuiButton-root"
              href="https://www.google.com/chrome/"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              target="_blank"
            >
              Chrome
            </a>
            <a
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1tvghku-MuiButtonBase-root-MuiButton-root"
              href="https://www.apple.com/safari/"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              target="_blank"
            >
              Safari
            </a>
            <a
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1tvghku-MuiButtonBase-root-MuiButton-root"
              href="https://www.microsoft.com/en-us/windows/microsoft-edge"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              target="_blank"
            >
              Edge
            </a>
          </div>
        </div>
      </span>
    </div>
    <div
      className="MuiSnackbarContent-action css-1kr9x0n-MuiSnackbarContent-action"
    >
      <button
        aria-label="close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-nmordm-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
